lib/ansible/module_utils/common/parameters.py

```diff
@@ -940,3 +926,27 @@ def remove_values(value, no_log_strings):
                     raise TypeError('Unknown container type encountered when removing private values from output')
 
     return new_value
+
+
+def __getattr__(importable_name):
+    """Inject import-time deprecation warnings."""
+    if importable_name in {
+        "binary_type", "text_type",
+        "integer_types", "string_types",
+        "PY2", "PY3",
+    }:
+        import importlib
+        importable = getattr(
+            importlib.import_module("ansible.module_utils.six"),
+            importable_name
+        )
+    else:
+        raise AttributeError(
+            f"Cannot import name {importable_name!r} from {__name__!r} ({__file__!s})"
+        )
+
+    deprecate(
+        msg=f"Importing {importable_name!r} from {__name__!r} is deprecated.",
+        version="2.23",
+    )
+    return importable
```

Comment:
Actually, this might be even cleaner:
```suggestion
    if importable_name not in {
        "binary_type", "text_type",
        "integer_types", "string_types",
        "PY2", "PY3",
    }:
        raise AttributeError(
            f"Cannot import name {importable_name!r} from {__name__!r} ({__file__!s})"
        )

    deprecate(
        msg=f"Importing {importable_name!r} from {__name__!r} is deprecated.",
        version="2.23",
    )

    import importlib
    return getattr(
        importlib.import_module("ansible.module_utils.six"),
        importable_name
    )
```

Reply:
I am not sure if eliminating one `else` branch makes it cleaner but it makes it harder to potentially add a deprecation for imports from different modules.